### ISSUE TITLE
Rails 4.1 views integration with Batman views

### ISSUE THREAD

**Issue by gauravtiwari** [zohararad/batsoe#1]
Hi @zohararad 

Thanks for the tutorial. 

I am trying to integrate Batman into my existing rails app with controller, models and views. 

I read your documentation for integrating views into Batman, but its still rendering blank page. Please, if you can add some information on how to add Batman into an existing rails app - it's working and has all relevant code?

Thank you for your time.

Gaurav

**Comment by zohararad** [comment]
@gauravtiwari I'm having a hard time figuring out where your problem lies. Is it just a matter of Rails not rendering Batman views on the server, or a matter of your Batman application not running?

What versions of Rails and Batman are you using? Maybe you can link to the source code?

**Comment by gauravtiwari** [comment]
Hi @zohararad I have basically followed your tutorial and setup everything, however, the helper function you created to create batman views is giving me errors- 

Forex: Unexpect ")" (this is coming from simple_form wrapper)
'f' is undefined  - in simple form views. I am using Simple_Form gem. 

Rails version: 4.1
Batman version - latest on the Repo

If you need more information, please let me know. 

Basically, here is what I want to do: I have an existing app (75% complete) and it follows a traditional rails app convention, however, I have now decided to use Batman for front end. So, what you think I shall do to integrate batman at this stage? Your solution seems viable for integrating both views as one - server side and client side. 

Please let me know if you have any questions. 

Regards,
Gaurav

**Comment by zohararad** [comment]
Ok, so it sounds to me like the problem is with parsing your Batman views.

Basically integrating Batman sounds like a good idea if you want to enjoy SEO + a single-page app.

What you need to do is to slowly migrate your existing rails views to Batman views, but since you're effectively writing Batman views, don't use rails helpers as they won't be parsed into HTML.

The idea is to write HTML with some dynamic ERB that gets cleaned up before its consumed by Batman.

The helper method that processes the views for Batman doesn't know how to process form helpers for example, and these helpers won't do you any good, as you actually need the HTML they create. I think this is the reason for the failure.

So, try writing HTML-based views, with no Rails view helpers and see if that solves the problem
